Extensions/xkit_patches.js

```diff
@@ -160,6 +160,22 @@ XKit.extensions.xkit_patches = new Object({
 			XKit.tools.Nx_XHR = function(details) {
 				details.timestamp = new Date().getTime() + Math.random();
 
+				const standard_headers = {
+					"X-Requested-With": "XMLHttpRequest",
+					"X-Tumblr-Form-Key": XKit.interface.form_key()
+				};
+
+				if (details.headers === undefined) {
+					details.headers = standard_headers;
+				} else {
+					let existing = Object.keys(details.headers).map(x => x.toLowerCase());
```

Comment:
I tested this out on Safari expecting it to shatter like an iPhone Xr but apparently they support all of this so :+1:

for extra credit both `existing` and `x` could be const

Reply:
not sure about this; my brain notices "this value is different in each function call" and "this is a different value per iteration" respectively, more than "this value isn't modified post-assignment" for both